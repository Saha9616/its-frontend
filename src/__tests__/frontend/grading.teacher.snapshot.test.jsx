import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import GradingDashboardView from "@/app/courses/[courseId]/[questionId]/grading/page";

// mock courseId param
const mockQuestionParams = {
  questionId: "mockQuestionId",
  courseId: "mock123",
};

// mock usePathname
jest.mock("next/navigation", () => ({
  usePathname: () => ({
    replace: (text1, text2) => text2,
  }),
}));

// mock server action getUserProps
jest.mock("@/actions/getUserProps", () => {
  return {
    getUserProps: jest.fn().mockResolvedValue({
      id: "mockId1",
      email: "mockEmail1",
      password: "mockPw1",
      role: "TEACHER",
      school_id: "mockSchoolId",
    }),
  };
});

// mock server action getQuestionSubmissions
jest.mock("@/actions/getQuestionSubmissions", () => {
  const mockUserSubmissions = [
    {
      id: "mockSubmissionId2",
      submit_time: "2024-04-08",
      submitted_program: "mockProgram2",
      feedback: "mockFeedback2",
      grade: "10",
      question_id: "mockQuestionId",
      user_id: "mockId2",
      email: "mockEmail2",
      role: "STUDENT",
      school_id: "mockSchoolId",
    },
  ];

  return {
    getQuestionSubmissions: jest.fn().mockResolvedValue(mockUserSubmissions),
  };
});

it("renders teacher's grading page unchanged", async () => {
  const result = await GradingDashboardView({ params: mockQuestionParams });
  const { container } = render(result);
  expect(container).toMatchSnapshot();
});
