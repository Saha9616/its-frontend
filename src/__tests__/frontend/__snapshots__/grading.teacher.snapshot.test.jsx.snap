// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders teacher's grading page unchanged 1`] = `
<div>
  <div
    class="flex justify-between items-center mb-4"
  >
    <h1
      class="text-xl font-semibold"
    >
      Grading Dashboard
    </h1>
    <a
      href="/courses/mock123"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
      >
        Return to course
      </button>
    </a>
  </div>
  <div
    class="rounded-md border"
  >
    <div
      class="relative w-full overflow-auto"
    >
      <table
        class="w-full caption-bottom text-sm"
      >
        <thead
          class="[&_tr]:border-b"
        >
          <tr
            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
          >
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Select
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Student Email
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Grades
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Feedback
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Action
            </th>
          </tr>
        </thead>
        <tbody
          class="[&_tr:last-child]:border-0"
        >
          <tr
            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
            data-state="false"
          >
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              <button
                aria-checked="false"
                aria-label="Select row"
                class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
                data-state="unchecked"
                role="checkbox"
                type="button"
                value="on"
              />
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              <span>
                mockEmail2
              </span>
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              10
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              mockFeedba...
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              <a
                href="mockSubmissionId2"
              >
                <svg
                  class="lucide lucide-chevron-right-circle text-muted-foreground"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                  <path
                    d="m10 8 4 4-4 4"
                  />
                </svg>
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
